diff --git a/meson.build b/meson.build
index 9559dd2ebb1..3c0cd63f225 100644
--- a/meson.build
+++ b/meson.build
@@ -1651,36 +1651,36 @@ _drm_nouveau_ver = '2.4.102'
 _drm_intel_ver = '2.4.75'
 _drm_ver = '2.4.109'
 
-_libdrm_checks = [
-  ['intel', with_gallium_i915],
-  ['amdgpu', (with_amd_vk and not with_platform_windows) or with_gallium_radeonsi],
-  ['radeon', (with_gallium_radeonsi or with_gallium_r300 or with_gallium_r600)],
-  ['nouveau', with_gallium_nouveau],
-]
-
-# Loop over the enables versions and get the highest libdrm requirement for all
-# active drivers.
-_drm_blame = ''
-foreach d : _libdrm_checks
-  ver = get_variable('_drm_@0@_ver'.format(d[0]))
-  if d[1] and ver.version_compare('>' + _drm_ver)
-    _drm_ver = ver
-    _drm_blame = d[0]
-  endif
-endforeach
-if _drm_blame != ''
-  message('libdrm @0@ needed because @1@ has the highest requirement'.format(_drm_ver, _drm_blame))
-endif
-
-# Then get each libdrm module
-foreach d : _libdrm_checks
-  if d[1]
-    set_variable(
-      'dep_libdrm_' + d[0],
-      dependency('libdrm_' + d[0], version : '>=' + _drm_ver)
-    )
-  endif
-endforeach
+#_libdrm_checks = [
+#  ['intel', with_gallium_i915],
+#  ['amdgpu', (with_amd_vk and not with_platform_windows) or with_gallium_radeonsi],
+#  ['radeon', (with_gallium_radeonsi or with_gallium_r300 or with_gallium_r600)],
+#  ['nouveau', with_gallium_nouveau],
+#]
+#
+## Loop over the enables versions and get the highest libdrm requirement for all
+## active drivers.
+#_drm_blame = ''
+#foreach d : _libdrm_checks
+#  ver = get_variable('_drm_@0@_ver'.format(d[0]))
+#  if d[1] and ver.version_compare('>' + _drm_ver)
+#    _drm_ver = ver
+#    _drm_blame = d[0]
+#  endif
+#endforeach
+#if _drm_blame != ''
+#  message('libdrm @0@ needed because @1@ has the highest requirement'.format(_drm_ver, _drm_blame))
+#endif
+#
+## Then get each libdrm module
+#foreach d : _libdrm_checks
+#  if d[1]
+#    set_variable(
+#      'dep_libdrm_' + d[0],
+#      dependency('libdrm_' + d[0], version : '>=' + _drm_ver)
+#    )
+#  endif
+#endforeach
 
 with_gallium_drisw_kms = false
 dep_libdrm = dependency(
